### PR TITLE
[xDS] Accept cpu_utilization over 100%

### PR DESCRIPTION
**Pull request by yousukseung**

### CODE REVIEW

File: include/grpcpp/ext/server_metric_recorder.h [hunk 1]
```diff
@@ -43,9 +43,10 @@ class ServerMetricRecorder {
...
   // Factory method. Use this to create.
   static std::unique_ptr<ServerMetricRecorder> Create();
-  /// Records the server CPU utilization in the range [0, 1].
-  /// Values outside of the valid range are rejected.
-  /// Overrides the stored value when called again with a valid value.
+  /// Records the server CPU utilization in the range [0, infy).
+  /// Values may be larger than 1.0 when the usage exceeds the reporter
+  /// dependent notion of soft limits. Values outside of the valid range are
+  /// rejected. Overrides the stored value when called again with a valid value.
   void SetCpuUtilization(double value);
```
Comment (markdroth) [inline include/grpcpp/ext/server_metric_recorder.h:50]: Please also update the comment on `CallMetricRecorder::RecordCpuUtilizationMetric()`:

https://github.com/grpc/grpc/blob/1267000bbb690353c7d5e05b32adcffdbbeed16c/include/grpcpp/ext/call_metric_recorder.h#L43

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 2]
```diff
@@ -282,6 +282,7 @@ class FakeResolverResponseGeneratorWrapper {
       response_generator_;
 };
 
+using xds::data::orca::v3::OrcaLoadReport;
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc:285]: Please move this up to line 87.

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 3]
```diff
@@ -2752,6 +2746,73 @@ TEST_F(ClientLbInterceptTrailingMetadataTest, BackendMetricDataMerge) {
...
+    ASSERT_TRUE(actual.has_value());
+    CheckLoadReportAsExpected(*actual, GetParam().received);
+  }
+  // Check LB policy name for the channel.
+  EXPECT_EQ("intercept_trailing_metadata_lb",
+            channel->GetLoadBalancingPolicyName());
+  EXPECT_EQ(kNumRpcs, num_trailers_intercepted());
+}
+
+INSTANTIATE_TEST_SUITE_P(
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc:2779]: I think using a parameterized test is overkill here.  It makes the code harder to understand, and it generates test names that do not actually help a human understand the purpose of the test.

Instead, please just add a helper method to the existing `ClientLbInterceptTrailingMetadataTest` framework to run a test with specified input and expected output, like this:

```
static void RunPerRpcMetricReportingTest(
    const OrcaLoadReport& reported,
    const OrcaLoadReport& expected) {
  // ...run test...
}
```

Then write a separate `TEST_F()` block for each test case you want to test, with an appropriate test name, which calls the helper function with appropriate values.  For example:

```
TEST_F(ClientLbInterceptTrailingMetadataTest, IgnoresOutOfRangeValues) {
  RunPerRpcMetricReportingTest(...);
}

TEST_F(ClientLbInterceptTrailingMetadataTest, AllowsCpuUtilizationGreaterThan1) {
  RunPerRpcMetricReportingTest(...);
}
```

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 4]
```diff
@@ -2345,30 +2345,66 @@ TEST_F(ClientLbPickArgsTest, Basic) {
...
+  }
+  OrcaLoadReportBuilder& SetUtilization(absl::string_view n, double v) {
+    (*report_.mutable_utilization())[n] = v;
+    return *this;
+  }
+  OrcaLoadReportBuilder& SetNamedMetrics(absl::string_view n, double v) {
+    (*report_.mutable_named_metrics())[n] = v;
+    return *this;
+  }
+  OrcaLoadReport Get() { return std::move(report_); }
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc:2381]: Please call this method `Build()`.

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 5]
```diff
@@ -3140,20 +3200,19 @@ TEST_F(WeightedRoundRobinTest, CallAndServerMetric) {
...
   request.mutable_param()->mutable_backend_metrics()->set_eps(
       std::numeric_limits<double>::min());
   request.mutable_param()->mutable_backend_metrics()->set_rps_fractional(100);
   ExpectWeightedRoundRobinPicks(DEBUG_LOCATION, stub,
-                                /*expected_weights=*/{3, 6, 4},
+                                /*expected_weights=*/{15, 10, 2},
                                 /*total_passes=*/3, &request);
   // Now send requests without per-call reported QPS.
-  // This should change WRR picks back to 1:2:4.
+  // This should change WRR picks back to 2:1:8.
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc:3213]: This comment doesn't match the numbers below.

Comment (yousukseung) [reply test/cpp/end2end/client_lb_end2end_test.cc]: Fixed.

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 4 above)
Comment (yousukseung) [reply test/cpp/end2end/client_lb_end2end_test.cc]: Done.

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 2 above)
Comment (yousukseung) [reply test/cpp/end2end/client_lb_end2end_test.cc]: Done.

File: include/grpcpp/ext/server_metric_recorder.h (hunk 1 above)
Comment (yousukseung) [reply include/grpcpp/ext/server_metric_recorder.h:50]: Done.

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 3 above)
Comment (markdroth) [reply test/cpp/end2end/client_lb_end2end_test.cc:2780]: Please refactor as I suggested.  I think the parameterized test interface here is much harder to read, and we simply don't need the complexity in this case.

Comment (yousukseung) [reply test/cpp/end2end/client_lb_end2end_test.cc]: Updated.

**Review by markdroth — APPROVED** [review]
Looks great!

**Pull request merged**